CHANGELOG.md

```diff
@@ -27,6 +27,7 @@ in a way that makes the pools suitable for use in parallel tests.
 * Added a `#[diesel::declare_sql_function]` attribute macro to easily define support for 
   multiple sql functions at once via an `extern "SQL"` block
 * Support `[print_schema] allow_tables_to_appear_in_same_query_config = "fk_related_tables"` to generate separate `allow_tables_to_appear_in_same_query!` calls containing only tables that are related through foreign keys. (Default: `"all_tables"`.) It is not possible to build queries using two tables that don't appear in the same `allow_tables_to_appear_in_same_query!` call, but that macro generates O(n²) rust code, so this option may be useful to reduce compilation time. ([#4333](https://github.com/diesel-rs/diesel/issues/4333))
+* Add `[print_schema] pg_domains_as_custom_types` parameter to generate custom types for [PostgreSQL domains](https://www.postgresql.org/docs/current/domains.html) that match given regex. (Default: `empty`.) This option allows an application to selectively give special meaning for the serialization/deserialization of these types, avoiding the default behavior of treating the domain as the underlying type. ([#4592](https://github.com/diesel-rs/diesel/discussions/4592))
```

Comment:
Default: empty : an empty regex matched any type so I would expect all domains to use custom types, which would be a breaking change?
This probably wants to be an option instead, with default None?

Reply:
I think the default makes sense if the parameter is a list (which it currently is), so I changed the changelog to better reflect that `[e76a0a3](https://github.com/diesel-rs/diesel/pull/4594/commits/e76a0a37b8c6c637c4775eeb04edc397126e897e)`.

In this sense, empty doesn't mean "empty regex"  which matches everything, it means "empty list of regex" which matches nothing and makes the feature non-breaking.

Does that make sense to you @Ten0 and @weiznich ?

It could be argued it should be an optional Regex since the user can use the regex to make a list with "|", but there are already other parameters (at least one) that is a list of regexes.

Reply:
That makes sense. Isn't "regexes" correct english?
Either way what do you think about writing "Default: `[]`"?

Reply:
@Ten0 [3169d54](https://github.com/diesel-rs/diesel/pull/4594/commits/3169d54160d61f31d8b19c42757d3048e00eb902) how about this?

Reply:
Ah I missed it was written "any regex in the given list".
Let's write "any of the regexes in the given list, Default: []" just for correct english and I think it will be super clear! 😊